### PR TITLE
Remove sim telarray native configuration files

### PR DESCRIPTION
**Pull request by orelgueta**
This PR removes our dependence on sim_telarray configuration files entirely. We only use simtools created configuration files from now on.

I had to skip the `testeff` tests because of this change since in the current version, `testeff` does not allow to provide a different configuration directory. A fix for that is already available. In addition, testeff requires a file which is currently unavailable in the DB (MR on gitlab is open). 

A couple of explicit additions were necessary because CORSIKA config does not write out configuration files. However, the atmospheric profile is necessary for all CORSIKA simulations, so now it is written out explicitly in the `corsika_config` step.

### CODE REVIEW

File: simtools/model/site_model.py [hunk 1]
```diff
@@ -30,6 +31,7 @@ def __init__(
         mongo_db_config,
         model_version,
         label=None,
+        array_model=None,
```
Comment (GernotMaier) [inline simtools/model/site_model.py:34]: Missing entry in docstring.

**Review by GernotMaier — APPROVED** [review]
Thanks for this. Tiny comment on a missing entry in a docstring. Approved.

File: simtools/model/site_model.py (hunk 1 above)
Comment (orelgueta) [reply simtools/model/site_model.py:36]: Thanks!

**Comment by orelgueta** [comment]
Thanks for the review. Was about to merge but after fixing the docstring I see errors in all tests like

```
Error: Unable to resolve action `sonarsource/sonarqube-scan-action@v3`, unable to find version `v3`
```

I assume they are temporary so I will try rerunning the tests a bit later and if they all work (except the linter one) I will merge.

**Pull request merged**